pkg/build/controller/controller.go

```diff
@@ -172,7 +172,7 @@ func (bc *BuildController) nextBuildPhase(build *buildapi.Build) error {
 		build.Status.Reason = buildapi.StatusReasonCannotCreateBuildPod
 		return fmt.Errorf("failed to create build pod: %v", err)
 	}
-
+	build.Annotations[buildapi.BuildPodNameAnnotation] = podSpec.Name
```

Comment:
looks like the normal build creation flow always populates annotations, but i'm guessing in some test scenarios we don't go through that flow.  check/initialization added.